src/azure-cli/azure/cli/command_modules/appconfig/_validators.py

```diff
@@ -147,3 +147,15 @@ def validate_filter_parameter(string):
         else:
             logger.warning("Ignoring filter parameter '%s' because parameter name is empty.", string)
     return result
+
+
+def validate_secret_identifier(namespace):
+    """ Validate the format of keyvault reference secret identifier """
+    from azure.keyvault.key_vault_id import KeyVaultIdentifier
+
+    identifier = getattr(namespace, 'secret_identifier', None)
+    try:
+        # this throws an exception for invalid format of secret identifier
+        KeyVaultIdentifier(uri=identifier)
+    except Exception as e:
```

Comment:
>Exception [](start = 11, length = 9)

Any specific type of exception this can throw? Since our error message is very specific to secret identifier value.

Reply:
Going through the keyvault SDK, it looks like they only throw ValueError and TypeError at this time. But I didn't find any official documentation which says they can't throw anything else in future. I could change our error message to clarify that the exception was thrown by keyvault. 

---
In reply to: [358527193](https://github.com/Azure/azure-cli/pull/11571#discussion_r358527193) [](ancestors = 358527193)